main.py

```diff
@@ -0,0 +1,241 @@
+        if month is present then it only iterate over the single month
+    """
+    def readDatadromFiles(self):
+        if self.month == None:
+            for month in months.values():
+                self.readDataFromSingleFile(month)
+                
+            if len(self.data) == 0:
+                raise Exception("No file is found")
+        else: 
+            self.readDataFromSingleFile(months[self.month])
+            if len(self.data) == 0:
+                raise Exception("No file is found")
+    
+    """
+    This function is make connection with the single file and then read the data line by line from the file
+    it read first line and set the column names or the fields name from the first line of the file and then
+    set it in the col_name to make dict
+    then it read the other lines one by one and then pick the field name from col_name and value from the 
+    temperature record (temp_record) and make the dict
+    
+    """    
+    def readDataFromSingleFile(self, month):
+        try :
+            with open(f'weather_reports/Murree_weather_{self.year}_{month}.txt', 'r') as file:
+                single_file_records = {}
+                single_line = file.readline()
+                _, *col_names = single_line.split(',')
+                single_date_record = {}
+                
+                while True:
+                    line = file.readline()
+                    if line:
+                        date, *temp_records = line.split(',')
+                        for index in range(len(temp_records)):
+                            single_date_record[col_names[index]] = temp_records[index]
+                            
+                        single_file_records.update({date: single_date_record})
+                        single_date_record = {}  
+                    else:
+                        break
+                self.data.append(single_file_records)
+        except : 
+            pass
+            
+        
+class CalculateTemperatureValues:
+    def __init__(self, file_obj):
+        self.data = file_obj
+        
+    def calculateTemperature(self, type_of_calculation):
+        if type_of_calculation == 'H':
+            return self._calculateTemperatureWithField('Max TemperatureC', True)
+        elif type_of_calculation == 'L': 
+            return self._calculateTemperatureWithField('Min TemperatureC', False)
+        elif type_of_calculation == 'Hu': 
+            return self._calculateTemperatureWithField('Max Humidity', True)
+        elif type_of_calculation == 'avg_L':
+            return self._calculateTemperatureWithField('Mean TemperatureC', False)
+        elif type_of_calculation == 'avg_H':
+            return self._calculateTemperatureWithField('Mean TemperatureC', True)
+        elif type_of_calculation == 'avg_Hu':
+            return self._calculateTemperatureWithField('Max Humidity', True)
```

Comment:
i think it is very complicated without the if/else chain because i use single function for all the functionalities. if i am not make if/else chain there then i need to check in the function it is calculated

Reply:
What if we use Dict here. Thought?